### PR TITLE
Riyad.setup nestjs

### PR DESCRIPTION
**Pull request by RiyadHassen**
implemented file upload and fixed merge conflicts.

### CODE REVIEW

File: starter_project_backend_nest/src/test/todo.spec.ts [hunk 1]
```diff
@@ -4,43 +4,63 @@ import { MongooseModule } from '@nestjs/mongoose';
...
-  });
-  describe('check todo exist', () => {
-    it('check if user exist', async () => {
-      const todo = await service.getTodos();
-      expect(todo).toBeDefined();
-    });
-  });
+let service: TodoService;
+let module: TestingModule;
+let usersercice: UserService;
```
Comment (Khalid-Sultan) [inline starter_project_backend_nest/src/test/todo.spec.ts:14]: Spelling needs a bit of a correction

File: starter_project_backend_nest/src/cloudinary/cloudinary.service.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import { CloudinaryProvider } from './cloudinary.provider';
+import toStream = require('buffer-to-stream');
+
+@Injectable()
+export class CloudinaryService {
+  async uploadImage(file: Express.Multer.File): Promise<any> {
+    return new Promise((resolve, reject) => {
+      const upload = cloudinary.uploader.upload_stream(
+        {
+          folder: 'starter_project',
```
Comment (Khalid-Sultan) [inline starter_project_backend_nest/src/cloudinary/cloudinary.service.ts:14]: extracting this value to an environment variable would be better I think

**Review by Khalid-Sultan — APPROVED** [review]
All looks good to me. Cloudinary looks clean

File: starter_project_backend_nest/src/cloudinary/cloudinary.module.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+@Module({
+  providers: [CloudinaryService, CloudinaryProvider],
+  exports: [CloudinaryProvider, CloudinaryService],
+  imports: [
+    ConfigModule,
+    MulterModule.registerAsync({
+      imports: [ConfigModule],
+      useFactory: async () => ({
+        fileFilter: (req, file, cb) => {
+          if (!file.mimetype.match(/png||jpeg||jpg||gif$i/)) {
```
[the remark this answers is not in the document]
Comment (RiyadHassen) [reply starter_project_backend_nest/src/cloudinary/cloudinary.module.ts:17]: sure

**Pull request merged**